Link `onMessageExternal` and `externally_connectable` docs (#19838)
mdn/content: files/en-us/mozilla/add-ons/webextensions/api/runtime/onmessageexternal/index.md

---
{{AddonSidebar()}}

Use this event to listen for messages from another extension or a website.

Whether the extension will allow messages from other extensions and websites is controlled by [`externally_connectable`](

> https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/externally_connectable) field in `manifest.json`.

Author: Is there a better way to link to this doc without a full URL?

Reviewer: The link should be relative e.g. `/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/externally_connectable`. But there isn't a macro or similar for simplifying links to the manifest key pages.

> Whether the extension will allow messages from other extensions and websites is controlled by [`externally_connectable`](
> https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/externally_connectable) field in `manifest.json`.

Reviewer: ```suggestion
Whether the extension allows messages from other extensions and websites is controlled by the [`externally_connectable`](/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/externally_connectable)  manifest key.
```

> Use this event to listen for messages from another extension or a website.
> 
> Whether the extension will allow messages from other extensions and websites is controlled by [`externally_connectable`](
> https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/externally_connectable) field in `manifest.json`.

Reviewer: Your change may give the impression that `externally_connectable` is required to enable communication and that communication from a website is a standard feature. They aren't. As the key is an experimental feature, I would suggest:

```suggestion
Use this event to listen for messages from another extension.

> **Note:** Using the [`externally_connectable`](/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/externally_connectable) manifest key, an [experimental feature](/en-US/docs/MDN/Writing_guidelines/Experimental_deprecated_obsolete#experimental) you can control which extension can communicate with yours and also enable websites to communicate with your extension. 
```

Reviewer 2: Let's not mark it as an experimental feature. It is supported by Chrome and Safari, and it looks like it will stick around.

There are however browser-specific notes that are worth calling out:

- `chrome.runtime.sendMessage`/`connect` in web pages is only available in Chrome.
- `browser.runtime.sendMessage`/`connect` in web pages is only available in Safari (not using `chrome.` because that was not possible because websites incorrectly assumed that the presence of `chrome` implied that the browser was Chrome).
- Neither API is supported in web pages in Firefox (https://bugzilla.mozilla.org/show_bug.cgi?id=1319168).
